include/openssl/core_numbers.h

```diff
@@ -58,6 +63,53 @@ OSSL_CORE_MAKE_FUNC(const OSSL_ITEM *,
 OSSL_CORE_MAKE_FUNC(int,core_get_params,(const OSSL_PROVIDER *prov,
                                          const OSSL_PARAM params[]))
 
+/* Memory allocation, freeing, clearing. */
+#define OSSL_FUNC_CORE_GET_CRYPTO_MALLOC       10
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_malloc, (size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_ZALLOC       11
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_zalloc, (size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_MEMDUP       12
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_memdup, (const void *str, size_t siz, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_STRDUP       13
+OSSL_CORE_MAKE_FUNC(char *,
+        CRYPTO_strdup, (const char *str, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_STRNDUP      14
+OSSL_CORE_MAKE_FUNC(char *,
+        CRYPTO_strndup, (const char *str, size_t s, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_FREE         15
+OSSL_CORE_MAKE_FUNC(void,
+        CRYPTO_free, (void *ptr, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_CLEAR_FREE   16
+OSSL_CORE_MAKE_FUNC(void,
+        CRYPTO_clear_free, (void *ptr, size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_REALLOC      17
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_realloc, (void *addr, size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_CLEAR_REALLOC 18
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_clear_realloc, (void *addr, size_t old_num, size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_SECURE_MALLOC 19
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_secure_malloc, (size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_SECURE_ZALLOC 20
+OSSL_CORE_MAKE_FUNC(void *,
+        CRYPTO_secure_zalloc, (size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_SECURE_FREE   21
+OSSL_CORE_MAKE_FUNC(void,
+        CRYPTO_secure_free, (void *ptr, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_SECURE_CLEAR_FREE 22
+OSSL_CORE_MAKE_FUNC(void,
+        CRYPTO_secure_clear_free, (void *ptr, size_t num, const char *file, int line))
+#define OSSL_FUNC_CORE_GET_CRYPTO_SECURE_MALLOC_INITIALIZED 23
```

Comment:
Is this one necessary?  It's not the providers concern if an application has started secure memory or not.

Reply:
I waffled on this, but I ultimately decided that a provider does have the right to know how it's keys are stored. But I don't have strong feelings either way.

Reply:
Thoughts @openssl/committers ?

I'm undecided too.  I'm leaning towards omitting it and adding it when and if it becomes desirable.  Primarily because a public interface can't be removed but a private one can be made public.

Reply:
I lean towards @paulidale's view.

Reply:
Okay, I removed it and pushed a new commit.

Reply:
Thanks.